### ISSUE TITLE
Add RPC to NS for generating ABP DevAddr

### ISSUE THREAD

**Issue by htdvisser** [TheThingsNetwork/lorawan-stack#294]
**Summary:** 

We should add an RPC to the NS that generates a DevAddr for ABP devices.

Refs https://github.com/TheThingsNetwork/lorawan-stack/issues/47#issuecomment-472831395, probably also related to #130.

**Why do we need this?**

Clients such as the CLI and Console need to request a DevAddr from the NS in order to register ABP devices with a properly peering-routable DevAddr (so not one from the experimental range).

**What is already there? What do you see now?**

The CLI and Console generate DevAddrs for ABP devices from the experimental range. This works fine on private networks, but won't work over peering.

The NS generates DevAddrs from a NetID (or prefix after #130).

**What is missing? What do you want to see?**

The NS should have functionality to allocate DevAddrs for ABP devices and expose this over an RPC.

**How do you propose to implement this?**

Basically the same as what it does for OTAA devices, and then exposed as an RPC. Inputs of the generator are ctx, NetID/DevAddrPrefix of the NS, AppID and perhaps (in the future) an approximate location (so that we can later make the DevAddr allocation and matching smarter/faster). It returns a DevAddr.

**What can you do yourself and what do you need help with?**

I can assist with the API definition and other discussion.

**Comment by htdvisser** [comment]
@rvolosatovs proposed alternative of generating a DevAddr if unset

**Comment by htdvisser** [comment]
@johanstokking says that it should be possible to bring your own DevAddr
